pkg/server/plugin/nodeattestor/awsiid/iid.go

```diff
@@ -328,13 +329,17 @@ func (p *IIDAttestorPlugin) SetLogger(log hclog.Logger) {
 }
 
 func (p *IIDAttestorPlugin) checkBlockDevice(instance ec2types.Instance) error {
-	ifaceZeroDeviceIndex := *instance.NetworkInterfaces[0].Attachment.DeviceIndex
-
-	if ifaceZeroDeviceIndex != 0 {
-		return fmt.Errorf("the EC2 instance network interface attachment device index must be zero (has %d)", ifaceZeroDeviceIndex)
+	ifaceZeroIndex := slices.IndexFunc(
+		instance.NetworkInterfaces,
+		func(net ec2types.InstanceNetworkInterface) bool {
+			return *net.Attachment.DeviceIndex == 0
+		},
+	)
+	if ifaceZeroIndex == -1 {
+		return fmt.Errorf("the EC2 instance network interface with device index 0 is inaccessible")
```

Comment:
No need for formatting the error.
```suggestion
		return errors.New("the EC2 instance network interface with device index 0 is inaccessible")
```